### PR TITLE
[events] Send digest header value to the location event

### PR DESCRIPTION
**Pull request by nagineni**
Send digest header value to location events in order to relate events which were generated with various configurations.

### CODE REVIEW

File: MapboxMobileEvents/MMEAPIClient.m [hunk 1]
```diff
@@ -207,6 +207,29 @@ - (void)processRequest:(NSURLRequest *)request {
             }
 
             NSUserDefaults.mme_configuration.mme_configUpdateDate = MMEDate.date;
+
+            NSString *digest = nil;
+            NSString *digestHeader = httpResponse.allHeaderFields[@"Digest"];
```
Comment (mr1sunshine) [inline MapboxMobileEvents/MMEAPIClient.m]: nit: Should we use here `valueForHTTPHeaderField` method?

**Review by mr1sunshine — APPROVED** [review]
LGTM with small nit.

File: MapboxMobileEvents/MMEAPIClient.m [hunk 2]
```diff
@@ -207,6 +207,29 @@ - (void)processRequest:(NSURLRequest *)request {
...
+            if (digestHeader) {
+                // Look for 'SHA-256' hash in case of multiple different digest values
+                NSArray *digestValues = [digestHeader componentsSeparatedByString:@","];
+                if (digestValues.count >= 2) {
```
Comment (frederoni) [inline MapboxMobileEvents/MMEAPIClient.m]: NSArray is now `key=value`, right?
could we do something like (with some additional checks):
```
    NSArray *kvs = [digest componentsSeparatedByString:@","];
    NSMutableDictionary *dict = [NSMutableDictionary dictionary];
    for (NSString *field in kvs) {
        NSArray *keyValue = [field componentsSeparatedByString:@"="];
        dict[keyValue[0]] = keyValue[1];
    }
```
to get a cleaner structure like:
```
{
    Digest = abcd;
    "SHA-256" = 1234;
}
```
or are there too many variations in the data?

File: MapboxMobileEvents/MMEAPIClient.m (hunk 1 above)
Comment (nagineni) [reply MapboxMobileEvents/MMEAPIClient.m]: 'valueForHTTPHeaderField:' is only available on iOS 13.0 or newer, so I am not sure whether we should use that or not.

File: MapboxMobileEvents/MMEAPIClient.m (hunk 2 above)
Comment (nagineni) [reply MapboxMobileEvents/MMEAPIClient.m]: Unfortunately, the current config service is not sending the `digest` header as `<key>=<value>` format, and that's the reason we need handle the different corner cases to fallback to the value(just hash) received from the config service.

**Pull request merged**